tools/validators/instance_validator/tests/telemetry_validator_report_test.py

```diff
@@ -0,0 +1,240 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for telemetry_validator_report.py."""
+
+import datetime
+from unittest import mock
+
+from absl.testing import absltest
+
+from validate import telemetry_validation_report as tvr
+from validate.constants import ENTITY_CODE
+from validate.constants import ENTITY_GUID
+from validate.constants import ERROR_DEVICES
+from validate.constants import EXPECTED_DEVICES
+from validate.constants import EXPECTED_POINTS
+from validate.constants import EXTRA_DEVICES
+from validate.constants import EXTRA_POINTS
+from validate.constants import INVALID_DIMENSIONAL_VALUES
+from validate.constants import MESSAGE_TIMESTAMP
+from validate.constants import MESSAGE_VERSION
+from validate.constants import MISSING_DEVICES
+from validate.constants import MISSING_POINTS
+from validate.constants import MISSING_PRESENT_VALUES
+from validate.constants import REPORT_TIMESTAMP
+from validate.constants import TELEMETRY_MESSAGE_ERRORS
+from validate.constants import TELEMETRY_MESSAGE_WARNINGS
+from validate.constants import TIMESTAMP_FORMAT
+from validate.constants import UNMAPPED_STATES
+
+
+TEST_GUID = 'b93304dc-1dce-4f8b-8f69-8e34b93fa69c'
+TEST_CODE = 'ABC_123'
+TEST_TIMESTAMP = 'test_timestamp'
+
+
+class TelemetryValidatorReportTest(absltest.TestCase):
+
+  @mock.patch.object(tvr, 'datetime')
+  def testGenerateReport(self, mock_datetime):
+    mock_datetime.datetime.now().strftime.return_value = TEST_TIMESTAMP
+    expected_points = [
+        tvr.TelemetryPoint(
+            point_name='return_water_temperature_sensor', present_value=100.0
+        ),
+        tvr.TelemetryPoint(
+            point_name='supply_water_temperature_sensor', present_value=200.0
+        ),
+    ]
+    missing_point = tvr.TelemetryPoint(
+        point_name='run_command', present_value=True
+    )
+    extra_point_1 = tvr.TelemetryPoint(
+        point_name='return_water_temperature_setpoint', present_value=100.0
+    )
+    extra_point_2 = tvr.TelemetryPoint(
+        point_name='supply_water_temperature_setpoint', present_value=200.0
+    )
```

Comment:
some variables are reused for the tests, can we create a todo or refactor?

Reply:
Done.